src/user_lists/views.py

```diff
@@ -53,3 +63,155 @@ def destroy(self, request, *args, **kwargs):
         instance.items.filter(is_removed=False).delete()
         return Response({"success": True}, status=status.HTTP_200_OK)
 
+    @action(detail=False, methods=["get"], url_path="user_check")
+    def user_check(self, request):
+        user_lists = self.get_queryset().prefetch_related("items")
+        
+        lists_data = []
+        for list_obj in user_lists:
+            items = list_obj.items.filter(is_removed=False).order_by("-created_date")
+            items_data = [
+                {
+                    "id": item.id,
+                    "unified_document_id": item.unified_document_id,
+                }
+                for item in items
+            ]
+            
+            lists_data.append({
+                "id": list_obj.id,
+                "name": list_obj.name,
+                "is_public": list_obj.is_public,
+                "items": items_data,
+            })
+        
+        response_data = {"lists": lists_data}
+        serializer = UserCheckResponseSerializer(response_data)
```

Comment:
I think we should have an UserListOverviewSerializer that handles the logic depicted in this endpoint (the loops and construction of `items_data` should be in the serializer - it is the serializer's job to serialize the data!